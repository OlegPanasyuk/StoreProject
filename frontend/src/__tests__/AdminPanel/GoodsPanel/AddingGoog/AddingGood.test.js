import React from 'react';
import AddingGoodConnected, { AddingGood } from '../../../../AdminPanel/GoodsPanel/AddingGood';
import { shallow, configure, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

configure({ adapter: new Adapter() });

function setupComponent(props = {
    onHide: () => { },
    addErrorToState: () => { },
}) {
    let component = shallow(<AddingGood {...props} />);
    return { component, props };
}

describe('AddingGood --- Snapshot', () => {
    it('component should be rendered', () => {
        let { component } = setupComponent();
        expect(AddingGood.propTypes.onHide).toBe(PropTypes.func);
        expect(shallowToJson(component)).toMatchSnapshot();
    });
});

describe('AddingGood', () => {
    let initialState = {
       
    };

    const mockStore = configureStore();
    let store, wrapper;
    beforeEach(() => {
        store = mockStore(initialState);
        wrapper = mount(<Provider store={store}><AddingGoodConnected /></Provider>);
    });
    it('+++ render the connected(SMART) component',() => {
        expect(wrapper.find(AddingGoodConnected).length).toEqual(1);
    });
});