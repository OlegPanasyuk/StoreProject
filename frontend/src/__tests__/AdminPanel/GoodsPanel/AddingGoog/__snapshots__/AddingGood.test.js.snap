// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddingGood --- Snapshot component should be rendered 1`] = `
<Bootstrap(Modal)
  centered={true}
  onHide={[Function]}
  show={true}
>
  <ModalHeader
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle>
      Adding Goods
    </ModalTitle>
  </ModalHeader>
  <ModalBody>
    <Form
      as="form"
      inline={false}
    >
      <FormGroup
        as="div"
      >
        <FormLabel
          column={false}
          srOnly={false}
        >
          Name
        </FormLabel>
        <FormControl
          as="input"
          onChange={[Function]}
          type="text"
          value=""
        />
      </FormGroup>
      <FormGroup
        as="div"
      >
        <FormLabel
          column={false}
          srOnly={false}
        >
          Description
        </FormLabel>
        <FormControl
          as="textarea"
          onChange={[Function]}
          type="text"
          value=""
        />
      </FormGroup>
      <FormGroup
        as="div"
      >
        <FormLabel
          column={false}
          srOnly={false}
        >
          Price
        </FormLabel>
        <FormControl
          as="input"
          onChange={[Function]}
          type="text"
          value={0}
        />
      </FormGroup>
      <FormGroup
        as="div"
      >
        <FormLabel
          column={false}
          srOnly={false}
        >
          Group
        </FormLabel>
        <FormControl
          as="select"
          defaultValue="Open this select menu"
          onChange={[Function]}
        >
          <option
            disabled={true}
          >
            Open this select menu
          </option>
        </FormControl>
      </FormGroup>
    </Form>
  </ModalBody>
  <ModalFooter>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="light"
    >
      Clear
    </Button>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="primary"
    >
      Save changes
    </Button>
    <Overlay
      placement="right"
      rootClose={false}
      show={false}
      target={null}
      transition={[Function]}
    >
      <Component />
    </Overlay>
  </ModalFooter>
</Bootstrap(Modal)>
`;
